sql-plugin/src/main/scala/com/nvidia/spark/rapids/aggregate.scala

```diff
@@ -774,31 +772,35 @@ class GpuHashAggregateIterator(
   }
 
   /**
-   * Compute the aggregations on the projected input columns.
+   * Compute the aggregations on the projected input columns, and close input batch.
    * @param toAggregateBatch input batch to aggregate
-   * @param helper an internal object that carries state required to execute computeAggregate from
+   * @param helper an internal object that carries state required to execute the aggregate from
    *               different parts of the codebase.
    * @return aggregated batch
    */
-  private def computeAggregate(
-      toAggregateBatch: ColumnarBatch, helper: AggHelper): ColumnarBatch  = {
+  private def computeAggregateAndClose(
+      toAggregateBatch: ColumnarBatch, helper: AggHelper): ColumnarBatch = {
     val computeAggTime = metrics.computeAggTime
     val opTime = metrics.opTime
     withResource(new NvtxWithMetrics("computeAggregate", NvtxColor.CYAN, computeAggTime,
       opTime)) { _ =>
       // a pre-processing step required before we go into the cuDF aggregate, in some cases
       // casting and in others creating a struct (MERGE_M2 for instance, requires a struct)
-      withResource(helper.preProcess(toAggregateBatch)) { preProcessed =>
-        val resultBatch = if (groupingExpressions.nonEmpty) {
+      val preProcessed = withResource(toAggregateBatch) { _ =>
+        helper.preProcess(toAggregateBatch)
+      }
+
+      val aggregated = withResource(preProcessed) { _ =>
+        if (groupingExpressions.nonEmpty) {
           helper.performGroupByAggregation(preProcessed)
         } else {
           helper.performReduction(preProcessed)
         }
-
-        // a post-processing step required in some scenarios, casting or picking
-        // apart a struct
-        helper.postProcess(resultBatch)
       }
+
+      // a post-processing step required in some scenarios, casting or picking
+      // apart a struct
+      helper.postProcess(aggregated)
```

Comment:
It feels rather inconsistent that `postProcess` closes the aggregated batch, but `preProcess` does not.  This is not new, but it might be nice to rename `postProcess` to indicate that it closes the batch.

Reply:
made neither close the batch, so the names stay the same. Pushing a diff.

Reply:
https://github.com/NVIDIA/spark-rapids/pull/6859/commits/cbbbf6b2dbd48ef18394ea85e6d11bf017e54fed